tableauserverclient/models/tableau_auth.py

```diff
@@ -1,41 +1,70 @@
-class TableauAuth(object):
+class Credentials:
+    def __init__(self, site_id=None, user_id_to_impersonate=None):
+        self.site_id = site_id or ""
+        self.user_id_to_impersonate = user_id_to_impersonate or None
+
+    @property
+    def credentials(self):
+        credentials = "Credentials can be username/password, Personal Access Token, or JWT"
+        +"This method returns values to set as an attribute on the credentials element of the request"
+
+    def __repr__(self):
+        display = "All Credentials types must have a debug display that does not print secrets"
+
+
+def deprecate_site_attribute():
+    import warnings
+
+    warnings.warn(
+        "TableauAuth(..., site=...) is deprecated, " "please use TableauAuth(..., site_id=...) instead.",
+        DeprecationWarning,
+    )
+
+
+# The traditional auth type: username/password
+class TableauAuth(Credentials):
     def __init__(self, username, password, site=None, site_id=None, user_id_to_impersonate=None):
```

Comment:
To make "site" even harder to use but be backwards compatible, we can make it kw-only

`def __init__(self, username, password, site_id=None, user_id_to_impersonate=None, *, site=None):`

Since site and site_id are the same thing and are the 3rd positional in both cases, it will continue to work for legacy code, or code that is using site as a kw-arg.

Reply:
will skip this and just delete it after creating 0.19